mage_ai/orchestration/pipeline_scheduler.py

```diff
@@ -109,7 +115,7 @@ def schedule(self, block_runs: List[BlockRun] = None) -> None:
         if PipelineType.STREAMING == self.pipeline.type:
             self.__schedule_pipeline()
         else:
-            if self.pipeline_run.all_blocks_completed():
+            if self.pipeline_run.all_blocks_completed(self.allow_blocks_to_fail):
```

Comment:
We should still update the pipeline run status to FAILED if there're failed blocks.